equinix/errors.go

```diff
@@ -98,7 +98,7 @@ func setMap(d *schema.ResourceData, m map[string]interface{}) error {
 			err = f(d, key)
 		} else {
 			if key == "router" {
-				d.Set("gateway", v)
+				err = d.Set("gateway", v)
```

Comment:
This condition shouldn't be here. Whatever is calling setMap with the expectation that `router` will be renamed to `gateway` should just use a `gateway` key instead of `gateway`.

This condition has the potential to affect any resource with a `router` field.

Reply:
Since this is followed by an additional `d.Set`, I'll assume the intent was to duplicate a value into both field names when only one field name was given. In that case, call setMap with both field names and the same value.

Reply:
You read the intention correctly. When I originally made this change I was brand new to the provider and saw this as the fix but it didn't actually achieve the result I was looking for. Calling setMap with the additional field is the correct method and is updated in the fabricMappingHelper after this comment.